File: poc/juneseok/src/main/java/com/dnd/poc/retrospective/config/OpenApiConfig.java
package com.dnd.poc.retrospective.config;

import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.info.Contact;
import io.swagger.v3.oas.models.info.Info;
import io.swagger.v3.oas.models.info.License;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
class OpenApiConfig {

    @Bean
    OpenAPI retrospectiveOpenApi() {
        return new OpenAPI()
                .info(
                        new Info()
                                .title("Retrospective AI API")
                                .version("0.0.1")
                                .description(
                                        "취준생의 감정 텍스트를 KPT(Keep/Problem/Try) + 응원 메시지로 구조화하는 AI 회고 도우미.")
                                .contact(
                                        new Contact()
                                                .name("DDD-13 WEBBB")
                                                .url(
                                                        "https://github.com/DDD-Community/DDD-13-WEBBB_BE"))
                                .license(new License().name("Internal PoC")));
    }
}

File: apps/api/src/main/kotlin/com/ogu/shared/config/OpenApiConfig.kt
package com.ogu.shared.config

import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Info
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class OpenApiConfig {
    @Bean
    fun openApi(): OpenAPI =
        OpenAPI().info(
            Info()
                .title("오구오구 API")
                .description("오구오구 백엔드 REST API")
                .version("v1"),
        )
}
